go/host/storage/hostdb/batch.go

```diff
@@ -207,6 +209,9 @@ func GetBatchByTx(db HostDB, txHash gethcommon.Hash) (*common.PublicBatch, error
 	}
 	extBatch, err := GetBatchBySequenceNumber(db, seqNo)
 	if err != nil {
+		if errors.Is(err, errutil.ErrNotFound) {
```

Comment:
`fmt.Errorf("failed to fetch ext batch - %w", err)` this produces a wrapped error such that you can still call errors.Is(ErrNotFound) on the thing it produces and it will still say true because it walks the chain of wrapped errors to check. So I'm not sure why we need to handle the ErrNotFound case separately?

Reply:
I think you're right so I've reverted that for now, was originally in response to tudors PR comment here https://github.com/ten-protocol/go-ten/pull/2584#discussion_r2262303654